### PR TITLE
Add Azure storage support

### PR DESCRIPTION
**Pull request by halamix2**
**Description**

Changes proposed in this pull request:

- Add Azure storage support

**Related issue(s)**

- #25

### CODE REVIEW

File: components/operator/internal/state/controller_configuration.go [hunk 1]
```diff
@@ -28,15 +33,27 @@ func sFnControllerConfiguration(_ context.Context, r *reconciler, s *systemState
...
 		{spec.HealthzLivenessTimeout, &instance.Status.HealthzLivenessTimeout, "Duration of health check", ""},
 		{registry.SecretName, &instance.Status.SecretName, "Name of secret with registry access data", ""},
+		storageField,
 	}
 
 	updateStatusFields(r.k8s, instance, fields)
 	return nil
 }
 
+func getStorageField(storage *v1alpha1.Storage, instance *v1alpha1.DockerRegistry) fieldToUpdate {
```
Comment (pPrecel) [inline components/operator/internal/state/controller_configuration.go:47]: I would suggest considering moving the functionality of storage configuration into a separate state function. I think the state function should describe the functionality of a single and small domain and the `storage` configuration looks like such a small domain. Of course, I don't want to force such change. This is a suggestion only.

**Review by pPrecel — COMMENTED** [review]
I'm not sure if new data under the `Storage` struct should not be fetched from a secret. This looks like sensitive data and should not be put as plaintext in the DockerRegistry CR.

Example:
```yaml
apiVersion: operator.kyma-project.io/v1alpha1
kind: DockerRegistry
metadata:
  name: default
  namespace: kyma-system
spec:
  storage:
    azure:
      secretName: azure-storage
---
apiVersion: v1
kind: Secret
metadata:
  name: azure-storage
  namespace: kyma-system
data:
  accountKey: "accountKey"
  accountName: "accountName"
  container: "container"
```

**Comment by halamix2** [comment]
I've moved "secret" to a Secret

File: components/operator/internal/state/controller_configuration.go (hunk 1 above)
Comment (halamix2) [reply components/operator/internal/state/controller_configuration.go:47]: I don't think this change is necessary, at least for now

**Review by pPrecel — APPROVED** [review]

**Pull request merged**